plugins/modules/elb_classic_lb_info.py

```diff
@@ -158,7 +158,12 @@
 @AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
 def list_elbs(connection, names):
     paginator = connection.get_paginator('describe_load_balancers')
-    load_balancers = paginator.paginate(LoadBalancerNames=names).build_full_result().get('LoadBalancerDescriptions', [])
+    try:
+        load_balancers = \
+            paginator.paginate(LoadBalancerNames=names).build_full_result().get('LoadBalancerDescriptions', [])
+    except is_boto3_error_code('LoadBalancerNotFound'):
+        return []
```

Comment:
I'm torn about the right behaviour here.

While I don't think we should allow the exception to surface to the user, we did fail to get the information (because the LB doesn't exist), and this is the behaviour that we've had since Ansible 2.0.  Switching it out to just returning an empty list could break playbooks: if people expect 'failed', explicitly catch and handle it, then we switch to returning success and an empty list and they expect there to be an entry if it succeeded...

I feel like we should define a consistent behaviour (probably returning an empty result like this) and try to clean up where we behave differently, but it should be a very conscious decision.  @jillr - Any thoughts?

Reply:
If we're just spitting a traceback at users now I'd say that's a bug and we should stop doing so.  I agree that generally we should be returning empty lists whenever an info module is NotFound.  I'd like to see integration tests to cover this case, though we don't have a test suite for elb_classic_lb_info right now.  What do you think @tremble about aliasing that into the elb_classic_lb tests and adding some coverage?

We definitely need a changelog if the behaviour changes though.

Reply:
+1, this was also discussed on IRC and several of the usual suspects agree that at the very least when we accept a list, returning an empty list is the right behaviour.